sdk/python/pulumi_docker/image.py

```diff
@@ -160,8 +165,9 @@ def check_tag(t: Optional[str]):
                 'registry_server': registry.server if registry else None
             }
 
-        image_args = ImageArgs(image_name, build, local_image_name, registry, skip_push)
-        image_data = pulumi.Output.from_input(image_args).apply(get_image_data)
+        image_data = pulumi.Output.all(image_name, build, local_image_name, registry, skip_push).apply(
+            lambda args: get_image_data(_ImageArgs(*args))
+        )
 
         self.image_name, self.registry_server = image_data['image_name'], image_data['registry_server']
```

Comment:
I get a typing error here, `self.registry_server` should be a `pulumi.Output[Optional[str]]` I think (that's what I did in my fork [here](https://github.com/rienafairefr/pulumi-docker/tree/fix-registry-push))